Mathlib/NumberTheory/LegendreSymbol/AddCharacter.lean

```diff
@@ -47,12 +48,30 @@ section Additive
 -- The domain and target of our additive characters. Now we restrict to a ring in the domain.
 variable {R : Type u} [CommRing R] {R' : Type v} [CommMonoid R']
 
+/-- The values of an additive character on a ring of positive characteristic are roots of unity. -/
+lemma val_mem_rootsOfUnity (φ : AddChar R R') (a : R) (h : 0 < ringChar R) :
+    (φ.val_isUnit a).unit ∈ rootsOfUnity (ringChar R).toPNat' R' := by
+  simp only [mem_rootsOfUnity, Nat.toPNat'_coe, h, ↓reduceIte, Units.ext_iff, CharP.cast_eq_zero,
+    Units.val_pow_eq_pow_val, IsUnit.unit_spec, ← map_nsmul_pow, nsmul_eq_mul, zero_mul,
+    map_zero_one, Units.val_one]
```

Comment:
```suggestion
  simp only [mem_rootsOfUnity', Nat.toPNat'_coe, h, ↓reduceIte, IsUnit.unit_spec, ← map_nsmul_pow,
    nsmul_eq_mul, CharP.cast_eq_zero, zero_mul, map_zero_one]
```

Reply:
I tried to replace `CharP.cast_eq_zero`by `Nat.cast_zero`, but here it is really necessary! :smile: